src/jail_rules.c

```diff
@@ -269,18 +270,18 @@ jail_rules_add_profile(
     const char* name,
     gboolean require)
 {
-    char* path;
-
-    if (g_str_has_suffix(name, SAILJAIL_PROFILE_SUFFIX)) {
-        path = g_build_filename(conf->perm_dir, name, NULL);
-    } else {
-        char* file = g_strconcat(name, SAILJAIL_PROFILE_SUFFIX, NULL);
-
-        path = g_build_filename(conf->perm_dir, file, NULL);
-        g_free(file);
+    char* path = 0;
+
+    /* Accept only file names with expected suffixes and
+     * no directory components. */
+    if (name && !strchr(name, '/')) {
```

Comment:
It's a static function, it's only invoked from 2 places and in both places `name` can't be `NULL`. There's no need to check it for `NULL` here.

Reply:
ack